### PR TITLE
[DOCS] Reformats update by query API.

### PR DESCRIPTION
**Pull request by debadair**
This PR updates the doc update by query API to align with the new API reference template.

Relates to elastic/docs#937 and #45365

### CODE REVIEW

File: docs/reference/docs/delete-by-query.asciidoc [hunk 1]
```diff
@@ -85,13 +85,12 @@ by query once the request completes. This is different than the delete API's
 request to be refreshed. Unlike the delete API, it does not support 
 `wait_for`.
 
-[[docs-delete-by-query-task-api]]
+[[docs-delete-by-query-====]]
```
Comment (jrodewig) [inline docs/reference/docs/delete-by-query.asciidoc]: Was this change intentional?
> [[docs-delete-by-query-====]]

**Review by jrodewig — APPROVED** [review]
LGTM. Few minor questions.

File: docs/reference/docs/delete-by-query.asciidoc (hunk 1 above)
Comment (debadair) [reply docs/reference/docs/delete-by-query.asciidoc]: Nope. Good catch.
